### PR TITLE
duti: Add new port

### PR DESCRIPTION
**Pull request by newtonne**
###### Type(s)

- [x] enhancement

###### Tested on

macOS 10.13.5
Xcode 9.4.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by newtonne** [comment]
Hmm... looks like the configured `-isysroot` dir (`/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk`) doesn't exist on macOS 10.11. Anyone know what the path should be? Alternatively, we might be able to remove the `-isysroot` arg and have it look in the default search paths.

File: sysutils/duti/Portfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+categories          sysutils
+license             public-domain
+maintainers         {gmail.com:newtonne.github @newtonne} openmaintainer
+platforms           darwin
+supported_archs     x86_64
+
+description         Select default apps for documents and URL schemes on macOS.
+
+long_description    ${description}
```
Comment (pmetzger) [inline sysutils/duti/Portfile]: This sounds somewhat interesting from `description`, could `long_description` explain it further?

The github page has:

duti is a command-line utility capable of setting default applications for various document types on macOS, using Apple's Uniform Type Identifiers (UTI). A UTI is a unique string describing the format of a file's content. For instance, a Microsoft Word document has a UTI of com.microsoft.word.doc. Using duti, the user can change which application acts as the default handler for a given UTI.

**Comment by kencu** [comment]
I thought I'd try a build on `10.6.8` and it automatically found the right SDK and built through without any issue. What's happening on `10.11`, I wonder?

It appears it is possible to force the SDK, something like this:
```
configure.args-append \
                    --with-macosx-sdk=${configure.sdkroot}
```

File: sysutils/duti/Portfile (hunk 1 above)
Comment (newtonne) [reply sysutils/duti/Portfile]: Yup. I've added this in my latest commit 👍

**Comment by newtonne** [comment]
Thanks, @kencu. I tried that but it doesn't seem to have done the trick. I don't think I'm using it correctly since it isn't actually being set:
```
Executing:  cd "/opt/local/var/macports/build/_Users_travis_build_macports_macports-ports_sysutils_duti/duti/work/duti-1.5.4" && ./configure --prefix=/opt/local --with-macosx-sdk= 
```

**Comment by kencu** [comment]
failed again. 
```
clang: warning: no such sysroot directory: '/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk'
```
I'll run a build tonight on 10.11 and see what is wrong here.

**Comment by newtonne** [comment]
What about pointing to the command-line tools SDK?

```
--with-macosx-sdk=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
```

**Comment by newtonne** [comment]
Nope. Looks like this is why configure.sdkroot isn't returning anything: https://github.com/macports/macports-base/blob/master/src/port1.0/portconfigure.tcl#L407

**Comment by kencu** [comment]
I think you're right -- the issue here appears to me to be in that portconfigure.tcl file. On 10.11, it is supposed to be returning this:
```
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk
```
perhaps we should commit this, and see what fails, then fix those issues.

**Comment by kencu** [comment]
let's go back to this:
```
configure.args-append --with-macosx-sdk=${configure.sdkroot}
```
and we'll commit that, and see what fallout we have.

**Comment by newtonne** [comment]
Thanks, @kencu 👍

**Comment by kencu** [comment]
Built on most systems. 10.11 succeeded, go figure. 10.10 failed. ${configure.sdkroot} is just passing in an empty string on all systems, so that turned out to be kinda useless. That's on us, though, not on your Port.

**Comment by newtonne** [comment]
10.11 _succeeded_? Was that on your own machine?

**Comment by kencu** [comment]
10.11 <https://build.macports.org/builders/ports-10.11_x86_64-builder/builds/62930/steps/install-port/logs/stdio>

10.10 <https://build.macports.org/builders/ports-10.10_x86_64-builder/builds/63335/steps/install-port/logs/stdio>

**Comment by newtonne** [comment]
Ah. I didn't even know that website existed! How do I search for builds of a particular port?

Looks like the 10.11 build bot is using Xcode 8, whereas the 10.11 Travis machine has Xcode 7.

**Comment by pmetzger** [comment]
@kencu It would be cool if we got more of this documented. :(

**Pull request merged**